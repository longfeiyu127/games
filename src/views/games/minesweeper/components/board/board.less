.c-Board {
  width: 360px;
  height: 360px;
  margin: 0 auto;
  .line {
    display: flex;
  }
}
