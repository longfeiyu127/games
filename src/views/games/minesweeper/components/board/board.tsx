import React from 'react'
import { connect } from 'react-redux'
import './board.less'
import Cell from '../cell/cell'
import { shutDownType, CellState } from '@/services/store/modules/games/minesweeper.ts'

interface Iprops {
  BoardState: any[]
  trampleCell: () => {}
}

export function Board({ BoardState, trampleCell }: Iprops) {
  return (
    <div className="c-Board">
      {BoardState.map((line: any[], i: number) => {
        return (
          <div className="line" key={i}>
            {line.map((item: any, j: number) => {
              if (item && item.state === CellState.shutDown) {
                return <Cell type={shutDownType} mark={item.mark} onClick={trampleCell} cols={i} rows={j} key={j} />
              }
              return (
                <Cell mark={item.mark} type={item ? item.type : ''} onClick={trampleCell} cols={i} rows={j} key={j} />
              )
            })}
          </div>
        )
      })}
    </div>
  )
}

const mapStateToProps = (state: any) => ({
  BoardState: state.minesweeper.BoardState
})

const mapDispatchToProps = (dispatch: any) => ({
  trampleCell: dispatch.minesweeper.trampleCell
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Board)
