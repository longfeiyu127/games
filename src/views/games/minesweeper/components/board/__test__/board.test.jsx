import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Board } from '../board.tsx'
import Cell from '../../cell/cell'
Enzyme.configure({ adapter: new Adapter() })

const testData = [
  [
    { type: '‚¨ú', cols: 0, rows: 0, state: 0, mark: false },
    { type: '1Ô∏è‚É£', cols: 0, rows: 1, state: 0, mark: false },
    { type: '1Ô∏è‚É£', cols: 0, rows: 2, state: 1, mark: true },
    { type: '1Ô∏è‚É£', cols: 0, rows: 3, state: 0, mark: false },
    { type: '‚¨ú', cols: 0, rows: 4, state: 0, mark: true },
    { type: '‚¨ú', cols: 0, rows: 5, state: 0, mark: false },
    { type: '‚¨õ', cols: 0, rows: 6, state: 0, mark: false },
    { type: '1Ô∏è‚É£', cols: 0, rows: 7, state: 0, mark: false },
    { type: '3Ô∏è‚É£', cols: 0, rows: 8, state: 1, mark: false },
    { type: 'üí£', cols: 0, rows: 9, state: 0, mark: false }
  ]
]

describe('<Board />', () => {
  it('component minesweeper/Board render', () => {
    const Dom = renderer.create(<Board BoardState={testData} trampleCell={() => {}} />).toJSON()
    expect(Dom).toMatchSnapshot()
  })

  it('component minesweeper/Board mount', () => {
    const onClick = jest.fn()
    const board = mount(<Board BoardState={testData} trampleCell={onClick} />)
    expect(board.find(Board).length).toBe(1)
    expect(board.find(Cell).length).toBe([].concat(...testData).length)
  })
})
