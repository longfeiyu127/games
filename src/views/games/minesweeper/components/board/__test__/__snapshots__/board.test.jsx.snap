// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> component minesweeper/Board render 1`] = `
<div
  className="c-Board"
>
  <div
    className="line"
  >
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬜
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      1️⃣
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬛
      <div
        className="mark"
      >
        🚩
      </div>
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      1️⃣
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬜
      <div
        className="mark"
      >
        🚩
      </div>
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬜
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬛
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      1️⃣
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      ⬛
    </div>
    <div
      className="c-Cell"
      onClick={[Function]}
    >
      💣
    </div>
  </div>
</div>
`;
